packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts

```diff
@@ -409,11 +440,25 @@ export class Metric implements IMetric {
     const stack = cdk.Stack.of(scope);
 
     return this.with({
-      region: cdk.Token.isUnresolved(stack.region) ? undefined : stack.region,
-      account: cdk.Token.isUnresolved(stack.account) ? undefined : stack.account,
+      stackAccount: cdk.Token.isUnresolved(stack.account) ? undefined : stack.account,
+      stackRegion: cdk.Token.isUnresolved(stack.region) ? undefined : stack.region,
```

Comment:
Why do we need to `stackAccount` and `stackRegion` here? Can we directly set `region` and `account`?